### ISSUE TITLE
Forecast one value for each sample (t to t+1)

### ISSUE THREAD

**Issue by luizfysouza** [Nixtla/neuralforecast#700]
I'd like to pass a time series with N samples as input and predict t+1 for each sample. How can I accomplish this?

In the parameters, I have only found a way to change the prediction horizon, but it appears that this method uses the history to predict t+1 only for the last time step, rather than for all the samples.

**Comment by cchallu** [comment]
Hi @luizfysouza. Can you provide more details, please? By sample, do you mean timestamp, right? So you want to predict rolling forecasts for the entire time series. The `predict` method is designed only to produce outsample forecasts (the future after the last observation of the input data). You can use the `cross_validation` method to produce rolling forecasts for the test set (specified with `n_windows` or `test_size`).

**Comment by luizfysouza** [comment]
> Hi @luizfysouza. Can you provide more details, please? By sample, do you mean timestamp, right? So you want to predict rolling forecasts for the entire time series. The `predict` method is designed only to produce outsample forecasts (the future after the last observation of the input data). You can use the `cross_validation` method to produce rolling forecasts for the test set (specified with `n_windows` or `test_size`).

Yes, I mean timetamp... Sorry, not very used to time series haha.
So, yes, I'd like to predict my target variable at each timestep using just the previous timestamp. So, if I pass, for example, a series with 100 values, it should return me the prediction for all the 100 values (the last one being in the future). That is, I'd like to shift my series one to the left, pass it as input, predict t+1 for each timestamp and compare to the original (not shifted) in each position.
In order to do that, I should use input_size = 1 and horizon=1, right?
However, when I pass the number of windows or test size as the number of timestamps, it returns  "Exception: No windows available for training"

**Comment by cchallu** [comment]
Yes, it should be `input_size=1` and `horizon=1`. Is there a reason you want only to use one value as input to forecast? You can have a larger input size and keep the `horizon=1`.

You need to have windows available for training. Usually, we split the whole available data into train and test sets (everything before a particular timestamp is training data). The models are trained on the train set and then evaluated on the test set. If you set `test_size=total_length`, then the train set is empty.

You have two options:
* Define a train/test split, and recover the forecasts only for the test set, using the `cross_validation` function.
* Train the model with `fit` and set `test_size=0`. Then, use the `predict_insample` to produce all the forecasts for your data. Note that these are insample predictions, so the forecasts should be very accurate.

**Comment by luizfysouza** [comment]
Ok, that's exactly what I wanted, thtank you. It makes sense, the question was a bit stupid, i'm not used to using a  lib that is so encapsulated like this one.

Another question: Is there an easy way to predict more than 1 time series? For example, predict Temprature, humidity and energy consumption. If yes, how do I do it? Because I think i can't define 'y' as a vector, like [Temp, Humidity, Energy]. The only option would be to use the 'unique_id' and  predict one by one?

**Comment by cchallu** [comment]
The library is building **global models**, which means that one model will be used to forecast all the `unique_id` time series from the dataset (also named cross-learning). Most models are univariate, so forecasts are independent between them. You can set different features using the `unique_id` column (in a long format).
